test/Table_Tests/src/Formatting/Parse_Values_Spec.enso

```diff
@@ -395,22 +394,59 @@ spec =
             c2.value_type.should_equal Value_Type.Integer
-            Problems.expect_warning Leading_Zeros c2
+            Problems.assume_no_problems c2
 
             c3 = c1.parse type=Value_Type.Integer format=(Data_Formatter.Value.with_number_formatting allow_leading_zeros=True)
             c3.to_vector . should_equal [0, 0, 0, 1, -1, 1, 0, 10, 12345, Nothing]
             Problems.assume_no_problems c3
 
+        Test.specify "should correctly parse integers in US formats" <|
+            cUS = Column.from_vector "ints" ["1", "000123", "-1234", "1234567", "123e456"]
+            pUS = cUS.parse type=Value_Type.Integer
+            pUS.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["123e456"]) pUS
+
+            cUS2 = Column.from_vector "ints" ["1", "000123", "-1,234", "1,234,567", "12,34,56"]
+            pUS2 = cUS2.parse type=Value_Type.Integer
+            pUS2.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["12,34,56"]) pUS2
+
+            cUS3 = Column.from_vector "ints" ["1", "(000,123)", "-1,234", "1,234,567", "12,34,56"]
+            pUS3 = cUS3.parse type=Value_Type.Integer
+            pUS3.to_vector . should_equal [1, -123, Nothing, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["-1,234", "12,34,56"]) pUS3
+
+            cUS4 = Column.from_vector "ints" ["$1234", "$1,234", "$1,234,567","-$1,234", "($1,234,567)"]
+            pUS4 = cUS4.parse type=Value_Type.Integer
+            pUS4.to_vector . should_equal [1234, 1234, 1234567, -1234, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["($1,234,567)"]) pUS4
+
+        Test.specify "should correctly parse integers in European formats" <|
+            cDE = Column.from_vector "ints" ["1", "000123", "-1.234", "1.234.567", "12.34.56"]
+            pDE = cDE.parse type=Value_Type.Integer
+            pDE.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["12.34.56"]) pDE
+
+            cFR = Column.from_vector "ints" ["€1", "€000123", "€-1 234", "€   1 234 567", "€ 12 34 56"]
+            pFR = cFR.parse type=Value_Type.Integer
+            pFR.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["€ 12 34 56"]) pFR
+
+            cSW = Column.from_vector "ints" ["1", "000123", "-1'234", "1'234'567", "(123'456)"]
+            pSW = cSW.parse type=Value_Type.Integer
+            pSW.to_vector . should_equal [1, 123, -1234, 1234567, Nothing]
+            Problems.expect_warning (Invalid_Format.Error "ints" Value_Type.Integer ["(123'456)"]) pSW
```

Comment:
I'm especially interested what will happen for:
```
c = Column.from_vector "X" ["1,000", "1.000"]
c.parse == ???
c.parse type=Value_Type.Integer == ???
c.parse type=Value_Type.Float == ???
```

This is a case that IMO has some possibility of confusion as it's really impossible to tell what format we are dealing with is it `1` and `1000` or the reverse? Ideally I think it would be good to have some warning if there is such ambiguity (it can ofc be resolved if there are additional entires like `"1,000,000.0"` or `"1.000,0`). But that may be a bit too complicated, so I don't want to add additional work that has little benefit.

But we should at least have it clearly documented (at least by these tests) what the expected current behaviour is.